### PR TITLE
SystemView fix & upgrade

### PR DESCRIPTION
**Pull request by cepetr**
This small PR fixes the integration of Segger SystemView into Trezor firmware.

1) Fixes the target code build (broken by the firmware split into kernel/app).
2) Upgrades the SystemView target code from to the latest version (3.20 -> 3.58).
3) Adds logging functionality to the interrupt handler code (I2C, SVCall, PendSV, DMA, Display TE interrupt).

Tested on STM32U5 (requires increasing `KERNEL_SRAM1_SIZE` to 20KB).

### CODE REVIEW

**Review by hiviah — CHANGES_REQUESTED** [review]
This won't build for me when `SYSTEM_VIEW` is enabled feature flags without this modification below:

* one file was moved between directories in the PR change
* `SOURCE_FIRMWARE` can't be appended to before it's defined (`scons` fails thus on `NameError: name 'SOURCE_FIRMWARE' is not defined`)

```diff
diff --git a/core/SConscript.kernel b/core/SConscript.kernel
index 8e0c6756b..9b08432c9 100644
--- a/core/SConscript.kernel
+++ b/core/SConscript.kernel
@@ -225,22 +225,6 @@ CPPDEFINES_MOD += [
 if FEATURE_FLAGS["RDI"]:
     CPPDEFINES_MOD += ['RDI']
 
-if FEATURE_FLAGS["SYSTEM_VIEW"]:
-    SOURCE_FIRMWARE += [
-        'embed/segger/SEGGER/SEGGER_SYSVIEW_Config_NoOS.c',
-        'embed/segger/SEGGER/SEGGER_SYSVIEW.c',
-        'embed/segger/SEGGER/SEGGER_RTT.c',
-        'embed/segger/SEGGER/SEGGER_RTT_ASM_ARMv7M.S',
-        'embed/segger/SEGGER/Syscalls/SEGGER_RTT_Syscalls_GCC.c',
-        'embed/firmware/systemview.c',
-    ]
-    CPPPATH_MOD += [
-        'embed/segger/SEGGER/',
-        'embed/segger/Config/',
-    ]
-    CPPDEFINES_MOD += ['SYSTEM_VIEW']
-    CCFLAGS_MOD += '-DSYSTEM_VIEW '
-
 TRANSLATION_DATA = [
     "translations/en.json",
     "translations/order.json",
@@ -273,6 +257,22 @@ if 'sd_card' in FEATURES_AVAILABLE:
 else:
     SDCARD = False
 
+if FEATURE_FLAGS["SYSTEM_VIEW"]:
+    SOURCE_FIRMWARE += [
+        'embed/segger/Config/SEGGER_SYSVIEW_Config_NoOS.c',
+        'embed/segger/SEGGER/SEGGER_SYSVIEW.c',
+        'embed/segger/SEGGER/SEGGER_RTT.c',
+        'embed/segger/SEGGER/SEGGER_RTT_ASM_ARMv7M.S',
+        'embed/segger/SEGGER/Syscalls/SEGGER_RTT_Syscalls_GCC.c',
+        'embed/firmware/systemview.c',
+    ]
+    CPPPATH_MOD += [
+        'embed/segger/SEGGER/',
+        'embed/segger/Config/',
+    ]
+    CPPDEFINES_MOD += ['SYSTEM_VIEW']
+    CCFLAGS_MOD += '-DSYSTEM_VIEW '
+
 env.Replace(
     CAT='cat',
     DD='dd',
```

File: core/embed/trezorhal/stm32f4/i2c_bus.c [hunk 1]
```diff
@@ -910,43 +911,55 @@ static void i2c_bus_er_handler(i2c_bus_t* bus) {
...
   i2c_bus_er_handler(&g_i2c_bus_driver[0]);
   mpu_restore(mpu_mode);
+  SEGGER_SYSVIEW_RecordExitISR();
 }
 #endif
 
 #ifdef I2C_INSTANCE_1
 void I2C_INSTANCE_1_EV_IRQHandler(void) {
-  mpu_mode_t mpu_mode = mpu_reconfig(MPU_MODE_DEFAULT);
+  SEGGER_SYSVIEW_RecordEnterISR mpu_mode_t mpu_mode =
```
[the remark this answers is not in the document]
Comment (cepetr) [reply core/embed/trezorhal/stm32f4/i2c_bus.c]: Yes, this is a type. Fixed here f52c288f9d479a9988ad90180c8f57085d0037d1

**Comment by cepetr** [comment]
> This won't build for me when `SYSTEM_VIEW` is enabled feature flags without this modification below:
> 
> * one file was moved between directories in the PR change
> * `SOURCE_FIRMWARE` can't be appended to before it's defined (`scons` fails thus on `NameError: name 'SOURCE_FIRMWARE' is not defined`)
> 
> ```diff
> diff --git a/core/SConscript.kernel b/core/SConscript.kernel
> index 8e0c6756b..9b08432c9 100644
> --- a/core/SConscript.kernel
> +++ b/core/SConscript.kernel
> @@ -225,22 +225,6 @@ CPPDEFINES_MOD += [
>  if FEATURE_FLAGS["RDI"]:
>      CPPDEFINES_MOD += ['RDI']
>  
> -if FEATURE_FLAGS["SYSTEM_VIEW"]:
> -    SOURCE_FIRMWARE += [
> -        'embed/segger/SEGGER/SEGGER_SYSVIEW_Config_NoOS.c',
> -        'embed/segger/SEGGER/SEGGER_SYSVIEW.c',
> -        'embed/segger/SEGGER/SEGGER_RTT.c',
> -        'embed/segger/SEGGER/SEGGER_RTT_ASM_ARMv7M.S',
> -        'embed/segger/SEGGER/Syscalls/SEGGER_RTT_Syscalls_GCC.c',
> -        'embed/firmware/systemview.c',
> -    ]
> -    CPPPATH_MOD += [
> -        'embed/segger/SEGGER/',
> -        'embed/segger/Config/',
> -    ]
> -    CPPDEFINES_MOD += ['SYSTEM_VIEW']
> -    CCFLAGS_MOD += '-DSYSTEM_VIEW '
> -
>  TRANSLATION_DATA = [
>      "translations/en.json",
>      "translations/order.json",
> @@ -273,6 +257,22 @@ if 'sd_card' in FEATURES_AVAILABLE:
>  else:
>      SDCARD = False
>  
> +if FEATURE_FLAGS["SYSTEM_VIEW"]:
> +    SOURCE_FIRMWARE += [
> +        'embed/segger/Config/SEGGER_SYSVIEW_Config_NoOS.c',
> +        'embed/segger/SEGGER/SEGGER_SYSVIEW.c',
> +        'embed/segger/SEGGER/SEGGER_RTT.c',
> +        'embed/segger/SEGGER/SEGGER_RTT_ASM_ARMv7M.S',
> +        'embed/segger/SEGGER/Syscalls/SEGGER_RTT_Syscalls_GCC.c',
> +        'embed/firmware/systemview.c',
> +    ]
> +    CPPPATH_MOD += [
> +        'embed/segger/SEGGER/',
> +        'embed/segger/Config/',
> +    ]
> +    CPPDEFINES_MOD += ['SYSTEM_VIEW']
> +    CCFLAGS_MOD += '-DSYSTEM_VIEW '
> +
>  env.Replace(
>      CAT='cat',
>      DD='dd',
> ```

fixed in this commit f52c288f9d479a9988ad90180c8f57085d0037d1

**Review by hiviah — APPROVED** [review]
OK builds and works now correctly with SystemView.

**Pull request merged**